files/scripts/supervisor-proc-exit-listener

```diff
@@ -62,52 +157,60 @@ def main(argv):
 
     critical_group_list, critical_process_list = get_critical_group_and_process_list()
 
+    process_under_alerting = {}
+    listener_ready = True
+
     while True:
         # Transition from ACKNOWLEDGED to READY
-        childutils.listener.ready()
-
-        line = sys.stdin.readline()
-        headers = childutils.get_headers(line)
-        payload = sys.stdin.read(int(headers['len']))
+        if listener_ready:
+            childutils.listener.ready()
```

Comment:
This is confusing. If the listener is ready we set the listener to READY? Do we even need to track this?

Maybe this pseudocode would be more straightforward:

```
# Transition from ACKNOWLEDGED to READY
childutils.listener.ready()

while True:
    select() on stdin
    if input received:
        handle input and then transition from READY to ACKNOWLEDGED (i.e., do what is currently on lines 185-212)

    Check if we need to log alerts (i.e., do what is on line 170-176)
```

Reply:
I like this idea!

Reply:
Sorry, I left that step out. My idea is as follows:

```
# Transition from ACKNOWLEDGED to READY
childutils.listener.ready()

while True:
    select() on stdin
    if input received:
        handle input and then transition from READY to ACKNOWLEDGED (i.e., do what is currently on lines 185-212)
        # Transition from ACKNOWLEDGED to READY
        childutils.listener.ready()

    Check if we need to log alerts (i.e., do what is on line 170-176)
```

Will there be any problems with calling `childutils.listener.ready()` and not immediately listening? If so, we could do this:

```
listener_state_ready = False
While True:
    if not listener_state_ready:
        childutils.listener.ready()
        listener_state_ready = True
    select() on stdin
    if input received:
        handle_input and do the transition from READY to ACKNOWLEDGED (childutils.listener.ok())
        listener_ready = False

   check if we need to log alerts
```

Reply:
I like the first one and it did work!